Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs

```diff
@@ -0,0 +1,317 @@
+            public PointerEventData eventDataRight;
+
+            public PointerData(IMixedRealityPointer pointer, EventSystem eventSystem)
+            {
+                this.pointer = pointer;
+                eventDataLeft = new PointerEventData(eventSystem);
+                eventDataMiddle = new PointerEventData(eventSystem);
+                eventDataRight = new PointerEventData(eventSystem);
+            }
+        }
+
+        /// <summary>
+        /// Mapping from pointer id to event data and click state
+        /// </summary>
+        protected readonly Dictionary<int, PointerData> pointerDataToUpdate = new Dictionary<int, PointerData>();
+
+        /// <summary>
+        /// List of pointers that need one last frame of updates to remove
+        /// </summary>
+        protected readonly List<PointerData> pointerDataToRemove = new List<PointerData>();
+
+        protected Camera RaycastCamera { get; private set; }
+
+        public override void ActivateModule()
+        {
+            base.ActivateModule();
+
+            RaycastCamera = MixedRealityToolkit.InputSystem?.FocusProvider.UIRaycastCamera;
+
+            MixedRealityToolkit.InputSystem?.Register(gameObject);
+        }
+
+        public override void DeactivateModule()
+        {
+            MixedRealityToolkit.InputSystem?.Unregister(gameObject);
+
+            RaycastCamera = null;
+
+            base.DeactivateModule();
+        }
+
+        public override bool ShouldActivateModule()
+        {
+            return pointerDataToUpdate.Count > 0 || pointerDataToRemove.Count > 0 || base.ShouldActivateModule();
+        }
+
+        /// <summary>
+        /// Process the active pointers from MixedRealityInputManager and all other Unity input.
+        /// </summary>
+        public override void Process()
+        {
+            CursorLockMode cursorLockStateBackup = Cursor.lockState;
+
+            try
+            {
+                // Disable cursor lock for MRTK pointers.
+                Cursor.lockState = CursorLockMode.None;
+
+                // Process pointer events as mouse events.
+                foreach(var p in pointerDataToUpdate)
+                {
+                    PointerData pointerData = p.Value;
+                    IMixedRealityPointer pointer = pointerData.pointer;
+
+                    if (pointer.IsInteractionEnabled
+                        && pointer.Rays != null
+                        && pointer.Rays.Length > 0
+                        && pointer.SceneQueryType == Physics.SceneQueryType.SimpleRaycast)
+                    {
+                        ProcessMouseEvent((int)pointer.PointerId);
+                    }
+                    else
+                    {
+                        // Remove Unity pointer if it exists, but leave the MRTK pointer in pointersData unless/until the pointer's source device is lost/disconnected.
```

Comment:
>// Remove Unity pointer if it exists, but leave the MRTK pointer in pointersData unless/until the pointer's source device is lost/disconnected. [](start = 24, length = 143)

Nit: This is a great comment, my only comment is it's probably more appropriate to have on the function itself, rather than here in the caller. The reason being, we probably don't wanna copy paste this for each time we call this, and also because this is an implementation detail of the function itself.

Reply:
Actually the behavior has changed a bit from the original version, so I added a new comment within the function. We're just processing a final mouse event in case the pointer is currently down and   invalidate the last mouse point.